### PR TITLE
DM-6282 Chart API: external API and the API for histogram

### PR DESCRIPTION
**Pull request by tgoldina**
External API for XYPlot and Histogram
- Added showHistogram
- added showXYPlot and showHistogram to external viewer interface
- the external charts can be tested from ffapi-highlevel-test (Show XYPlot and Show Histogram links)

- Also updated JSDoc, only public methods are included

### CODE REVIEW

File: src/firefly/js/charts/ui/ChartsContainer.jsx [hunk 1]
```diff
@@ -86,12 +86,19 @@ ChartsContainer.propTypes = {
...
 function ExpandedView(props) {
+    const {defaultCharts, closeable} = props;
     return (
         <div style={{ display: 'flex', height: '100%', flexGrow: 1, flexDirection: 'column', overflow: 'hidden'}}>
             <div style={{marginBottom: 3}}>
-                {props.closeable && <CloseButton style={{display: 'inline-block', paddingLeft: 10}} onClick={() => dispatchSetLayoutMode(LO_MODE.expanded, LO_VIEW.none)}/>}
+                {closeable && <CloseButton style={{display: 'inline-block', paddingLeft: 10}} onClick={() => dispatchSetLayoutMode(LO_MODE.expanded, LO_VIEW.none)}/>}
             </div>
-            <ChartsTableViewPanel expandedMode={true} expandable={false} {...props} />
+            <ChartsTableViewPanel expandedMode={true} expandable={false} chartType={'scatter'} {...props} />
```
Comment (loitly) [inline src/firefly/js/charts/ui/ChartsContainer.jsx]: This whole section of code is a simple cut/paste from standard view.  You should clean it up as you deemed necessary.  Although, it does seem a bit awkward to see so many components laid out, but only one get mounted.

File: src/firefly/js/templates/fireflyviewer/FireflyViewerManager.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ export function* layoutManager({title, views='tables | images | xyPlots'}) {
                 break;
 
             case TABLE_LOADED :
-                [showImages, images] = handleNewTable(action, images, showImages);
+                [showImages, images] = handleNewTable(action, images, showImages, showTables);
```
Comment (loitly) [inline src/firefly/js/templates/fireflyviewer/FireflyViewerManager.js]: I don't see an action that say an XyPlot is added to the results.  I only see a TABLE_LOADED action.  So, use TABLE_LOADED to track when XyPlot.  You may consider adding actions to define specific chart's life-cycle.

File: src/firefly/js/templates/fireflyviewer/FireflyViewerManager.js [hunk 3]
```diff
@@ -131,8 +133,9 @@ function handleLayoutChanges(action) {
     }
 }
 
-function handleNewTable(action, images, showImages) {
+function handleNewTable(action, images, showImages, showTables) {
```
Comment (loitly) [inline src/firefly/js/templates/fireflyviewer/FireflyViewerManager.js]: This saga is really confusing.  I plan to re-write it one day.  So, ignore the awkwardness for now.

**Comment by loitly** [comment]
Works fine.  Minor cleanup if you want to do it.  I leave that for you to decide.
Review is complete.

File: src/firefly/js/charts/ui/ChartsContainer.jsx (hunk 1 above)
Comment (tgoldina) [inline src/firefly/js/charts/ui/ChartsContainer.jsx]: Thank you. ChartsContainer will be completely rewritten in the next ticket.

**Pull request merged**